### PR TITLE
fix: [FX-2626] set up danger env variable for self-hosted runners

### PR DESCRIPTION
**Pull request by dmaklygin**
[FX-2626]

### Description

Self-hosted runners may override `GITHUB_URL` variable, so requests may be performed incorectly.

This code explains how danger defined API URL: https://github.com/danger/danger-js/blob/main/source/platforms/github/GitHubAPI.ts#LL452C96-L452C118

### How to test

- Use branch version in an existing GH Workflow with defined self-hosted runners

### Development checks

- [x] Add changeset according to [guidelines](https://github.com/toptal/davinci-github-actions/blob/master/_docs/contribution/changeset-guidelines.md) (if needed)

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>

[FX-2626]: https://toptal-core.atlassian.net/browse/FX-2626?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by dmaklygin** [comment]
@toptal-anvil ping reviewers

**Review by serbantarmure — APPROVED** [review]

File: danger/action.yml [hunk 1]
```diff
@@ -5,4 +5,7 @@ runs:
   using: composite
   steps:
     - shell: bash
+      env:
+        # We must specify the base URL for the GitHub API because the self-hosted runner overrides the default one
```
Comment (OleksandrNechai) [inline danger/action.yml:9]: We can see in the code what we do. So I would remove the text. But to me the reason is not yet obvious. How the fact that "The self-hosted runner overrides the default one"  relates to what we do? It overrides with incorrect one and we override it again? It overrides unless we specify this one? 🤷 
```suggestion
        # The self-hosted runner overrides the default one, so ...
```

**Review by OleksandrNechai — APPROVED** [review]
Code looks good, but the texting could be improved 👍

File: danger/action.yml (hunk 1 above)
Comment (dmaklygin) [reply danger/action.yml:9]: Yes, it overrides to incorrect one, and may use this env variable for another purpose.
I think, it was a cause why Danger added one more variable `DANGER_GITHUB_API_BASE_URL` and it has higher priority. I would rely on this env variable rather than common one: `GITHUB_URL`

**Review by TomasSlama — APPROVED** [review]

File: .changeset/update.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+---
+'davinci-github-actions': patch
+---
+
+---
+
+- update danger environment variable for proper API domain requests
```
[the remark this answers is not in the document]
Comment (dmaklygin) [reply .changeset/update.md:7]: Sure, I'll reword it.

File: danger/action.yml [hunk 3]
```diff
@@ -5,4 +5,6 @@ runs:
   using: composite
   steps:
     - shell: bash
+      env:
+        DANGER_GITHUB_API_BASE_URL: https://api.github.com
```
Comment (OleksandrNechai) [inline danger/action.yml:9]: Why did you remove the comment?
